### ISSUE TITLE
question regrading data

### ISSUE THREAD

**Issue by KodamaXYZ** [jiny2001/dcscn-super-resolution#64]
Hi, i want to train this model adding per pixel lable map for example a cityscape set image after semantic segmentation: 
![frankfurt_000000_000294_leftimg8bit](https://user-images.githubusercontent.com/45035487/50542254-d0baaf80-0bc1-11e9-8924-f4d2e69f5653.png)

my will is to create a new model using the lable map as an additional information + the regular image. i would love it if you could asist me on how to add the extra information to the training and evaluating steps. would help me alot in my project. Thanks.

**Comment by jiny2001** [comment]
Hi, I believe you can easily extend my (or any other) SR model.

My model's input and output dimension is like [N x H x W x C] and C is like 1(mono) or 3(color) in usual case.
You can change it as C = 3 + (number of labels). I mean, input/output channels for label can be one-hot vector if the number of kinds of label is around 10 or 20. The one bothering thing I think is, you should use multiple loss functions and balance it. For ex, you will use MSE for image part and will use cross entropy for label part. Thx.

**Comment by KodamaXYZ** [comment]
Hi @jiny2001  thanks so much for the answer could you point me in the right section of the code?

**Comment by jiny2001** [comment]
Uhm,,,
That's not only 1 or some lines you need to change. I believe you have to change at least 1) input/GT data loading/feeding part and 2) adjust some parts in DL network, input/output/upscale filters. 3) add softmax to get label part of super resolution result? and then 4) modify loss function.

Those parts are in DCSCN.py and helper/loader.py but I can't simply point it out because you have to change many lines. Hope those advise helps you. Thx.

**Comment by KodamaXYZ** [comment]
@jiny2001 awesome ! thanks for the help i will try to implement these changes :)

**Comment by jiny2001** [comment]
Sure! Please let me know when you have any question for detail!
